src/components/scenes/RequestScene.js

```diff
@@ -274,22 +275,44 @@ export class RequestComponent extends React.Component<Props, State> {
   onError = (errorMessage?: string) => this.setState({ errorMessage })
 
   render() {
-    const { currencyIcon, exchangeSecondaryToPrimaryRatio, guiWallet, primaryCurrencyInfo, secondaryCurrencyInfo, theme } = this.props
+    const { currencyIcon, exchangeSecondaryToPrimaryRatio, edgeWallet, guiWallet, primaryCurrencyInfo, secondaryCurrencyInfo, theme } = this.props
     const styles = getStyles(theme)
 
-    if (guiWallet == null || primaryCurrencyInfo == null || secondaryCurrencyInfo == null || exchangeSecondaryToPrimaryRatio == null) {
+    if (guiWallet == null || primaryCurrencyInfo == null || secondaryCurrencyInfo == null || exchangeSecondaryToPrimaryRatio == null || edgeWallet == null) {
       return <ActivityIndicator color={theme.primaryText} style={styles.loader} size="large" />
     }
 
     const requestAddress = this.props.useLegacyAddress ? this.state.legacyAddress : this.state.publicAddress
     const flipInputHeaderText = guiWallet ? sprintf(s.strings.send_to_wallet, guiWallet.name) : ''
     const { keysOnlyMode = false } = getSpecialCurrencyInfo(primaryCurrencyInfo.displayCurrencyCode)
 
+    // Balance
+    const nativeBalance = edgeWallet.balances[primaryCurrencyInfo.displayCurrencyCode] ?? '0'
```

Comment:
You need to use this function to get the balance: https://github.com/EdgeApp/edge-react-gui/blob/07682f7fe45e6f32641fb9e0bece11b0d77462fa/src/util/CurrencyWalletHelpers.js#L23

Otherwise you would include the stacked as well and have the same bug I fixed here: https://github.com/EdgeApp/edge-react-gui/commit/87b6576031d582f313189744bf2f19fc0859b6e8